infra/AuthStack.py

```diff
@@ -171,6 +171,89 @@ def _get_client_secret(
         return describe_cognito_user_pool_client.get_response_field(
             "UserPoolClient.ClientSecret"
         )
+    
+    def _create_policy(self) -> iam.PolicyDocument:
+        """
+        Create a policy for the KMS key to restrict access to the service.
+        """
+        return iam.PolicyDocument(
+            statements=[
+                iam.PolicyStatement(
+                    actions=[
+                        "kms:Encrypt",
+                        "kms:Decrypt",
+                        "kms:ReEncrypt*",
+                        "kms:CreateGrant",
+                        "kms:DescribeKey"
+                    ],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal("*")],
+                    conditions={
+                    "StringEquals": {
+                        "kms:CallerAccount": Stack.of(self).account,
+                        "kms:ViaService": f"secretsmanager.{Stack.of(self).region}.amazonaws.com",
+                    }
+                    },
+                ),
+                iam.PolicyStatement(
+                    actions=["kms:GenerateDataKey*"],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal("*")],
+                    conditions={
+                    "StringEquals": {
+                        "kms:CallerAccount": Stack.of(self).account,
+                        "kms:ViaService": "secretsmanager.*.amazonaws.com",
+                    }
+                    },
+                ),
+                iam.PolicyStatement(
+                    actions=[
+                        "kms:Describe*",
+                        "kms:Get*",
+                        "kms:List*",
+                        "kms:RevokeGrant"
+                    ],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal(f"arn:aws:iam::{Stack.of(self).account}:root")],
+                ),
+                iam.PolicyStatement(
+                    actions=["kms:Decrypt", "kms:DescribeKey"],
+                    effect=iam.Effect.ALLOW,
+                    resources=["*"],
+                    principals=[iam.ArnPrincipal("arn:aws:iam::884094767067:role/MAAP-ADE-K8S")],
```

Comment:
It would be helpful to set this as a constant with a descriptive name at the top.

Reply:
Ran through it a little fast, I'll add it to the env!